### PR TITLE
FISH-6280 CDI 4.0 Support

### PR DESCRIPTION
**Pull request by kalinchan**
Ran against Payara6. 

[mvn.test] [ERROR] Tests run: 4230, Failures: 1631, Errors: 0, Skipped: 2599
 [mvn.test] [INFO]
 [mvn.test] [INFO] ------------------------------------------------------------------------
 [mvn.test] [INFO] BUILD FAILURE
 [mvn.test] [INFO] ------------------------------------------------------------------------
 [mvn.test] [INFO] Total time:  03:20 min
 [mvn.test] [INFO] Finished at: 2022-05-16T17:20:57+01:00
 [mvn.test] [INFO] ------------------------------------------------------------------------

# To run
* Clone https://github.com/eclipse-ee4j/glassfish-cdi-porting-tck and build
* Set `porting.home` in build.properties to CDI TCK Glassfish Porting directory 
* Run `ant test`

### CODE REVIEW

File: build.properties [hunk 1]
```diff
@@ -33,7 +33,7 @@ impl.vi=glassfish
 #####################################
 # Glassfish V4 properties
 #####################################
-glassfish.home=/src/payara/Payara/appserver/distributions/payara/target/stage/payara6/glassfish
+glassfish.home=/home/kalin/git/Payara/appserver/distributions/payara/target/stage/payara6/glassfish
```
Comment (Pandrex247) [inline build.properties]: 🤔

File: build.xml [hunk 2]
```diff
@@ -118,268 +97,267 @@
...
                 <echo message="Java Home:  ${java.home}"/>
                    
                 <java outputproperty="sigoutput" fork="true" jar="${lib}/sigtest.jar">
-			<arg value="Signaturetest"/>
+            <arg value="Signaturetest"/>
                         <arg value="-Classpath"/>
                         <arg value="${java.home}/${java.classes.for.sig.ext}${pathsep}${glassfish.home}/modules/weld-osgi-bundle.jar${pathsep}${glassfish.home}/modules/jakarta.enterprise.cdi-api.jar${pathsep}${glassfish.home}/modules/jakarta.inject-api.jar${pathsep}${glassfish.home}/modules/jakarta.el.jar/${pathsep}${glassfish.home}/modules/jakarta.interceptor-api.jar${aix.jars}${pathsep}${glassfish.home}/modules/cdi-api-fragment.jar"/>
                         <arg value="-Package"/>
-                        <arg value="jakarta.enterprise"/>
+                        <arg value="javax.enterprise"/>
```
Comment (Pandrex247) [inline build.xml]: jakarta to javax? 🤔

File: glassfish-tck-runner/pom.template.xml [hunk 3]
```diff
@@ -343,6 +347,16 @@
             <id>payara-remote</id>
 
             <dependencies>
+                <dependency>
+                    <groupId>org.glassfish.jersey.inject</groupId>
+                    <artifactId>jersey-hk2</artifactId>
+                    <version>3.0.2</version>
```
Comment (Pandrex247) [inline glassfish-tck-runner/pom.template.xml]: Not the version we use in Payara? `3.0.3.payara-p2`

Comment (kalinchan) [reply glassfish-tck-runner/pom.template.xml]: Didn't seem to make any difference, I changed the version anyway.

File: glassfish-tck-runner/pom.template.xml [hunk 4]
```diff
@@ -21,12 +21,12 @@
     <parent>
         <artifactId>weld-core-parent</artifactId>
         <groupId>org.jboss.weld</groupId>
-        <version>4.0.1.SP1</version>
+        <version>5.0.0.CR2</version>
```
Comment (Pandrex247) [inline glassfish-tck-runner/pom.template.xml]: Can be `5.0.0.SP1`

Comment (Pandrex247) [inline glassfish-tck-runner/pom.template.xml]: Can be `5.0.0.SP1`

**Review by Pandrex247 — CHANGES_REQUESTED** [review]

**Comment by kalinchan** [comment]
Current status with review changes:

[mvn.test] [ERROR] Tests run: 1871, Failures: 1451, Errors: 0, Skipped: 40
 [mvn.test] [INFO]
 [mvn.test] [INFO] ------------------------------------------------------------------------
 [mvn.test] [INFO] BUILD FAILURE
 [mvn.test] [INFO] ------------------------------------------------------------------------
 [mvn.test] [INFO] Total time:  23:47 min
 [mvn.test] [INFO] Finished at: 2022-05-24T13:29:14+01:00
 [mvn.test] [INFO] ------------------------------------------------------------------------

**Review by Pandrex247 — APPROVED** [review]

File: glassfish-tck-runner/pom.template.xml [hunk 5]
```diff
@@ -42,10 +42,11 @@
         <htmlunit.version>2.9</htmlunit.version>
         <shrinkwrap.version>1.0.1</shrinkwrap.version>
         <minimum.maven.version>3.0.5</minimum.maven.version>
-        <cdi.api.version>3.0.0</cdi.api.version>
-        <cdi.tck.version>3.0.3</cdi.tck.version>
+        <cdi.api.version>4.0.0</cdi.api.version>
+        <cdi.tck.version>4.0.1</cdi.tck.version>
         <httpcomponents.version>4.5.5</httpcomponents.version>
         <arquillian.version>1.7.0.Alpha9</arquillian.version>
```
Comment (breakponchito) [inline glassfish-tck-runner/pom.template.xml]: Can we move to the aquillian version to 1.7.0.Alpha10 ? I saw the CDI TCK release documentation saying that this is the version to use

Comment (Pandrex247) [reply glassfish-tck-runner/pom.template.xml]:
```suggestion
        <arquillian.version>1.7.0.Alpha10</arquillian.version>
```

**Review by breakponchito — APPROVED** [review]
LGTM

**Pull request merged**